libs/types-rs/src/geometry.rs

```diff
@@ -305,6 +311,16 @@ impl Rect {
         }
     }
 
+    #[must_use]
+    pub fn overlaps(&self, other: &Self) -> bool {
+        let left = self.left.max(other.left);
+        let top = self.top.max(other.top);
+        let right = self.right().min(other.right());
+        let bottom = self.bottom().min(other.bottom());
+
+        left <= right && top <= bottom
+    }
+
```

Comment:
They subtract 1 because `Rect` is intended to be used with pixel values. `bottom()` and `right()` are meant to be the last values within the rect on the bottom and right, respectively. This makes more sense with integers vs floats. I know it seems weird to have `top()` and `bottom()` be the same for a `height: 1` rect, but that's what you end up with if you want `left()`, `top()`, `right()`, and `bottom()` to all be inclusive.